### PR TITLE
feature/cdp-youtube-scraper

### PR DESCRIPTION
**Pull request by dphoria**
### Link to Relevant Issue

This pull request implements the first version of `YoutubeIngestionScraper`. It has been tested to scrape minimal ingestion model data for Bellevue, WA.

### Description of Changes

Currently, the scraper is able to return YouTube video URLs as `Session.video_uri`.

```python
from cdp_scrapers.youtube_utils import YoutubeIngestionScraper
from datetime import datetime

begin = datetime(2023, 2, 5)
end = datetime(2023, 2, 10)
search_terms = "city council meeting"

s = YoutubeIngestionScraper(channel_name="BellevueTelevision", body_search_terms={"City Council":search_terms}, timezone="America/Los_Angeles")
events = s.get_events(begin=begin, end=end)
print(events[0])

# EventIngestionModel(
#     body=Body(name='City Council', ...),
#     sessions=[
#         Session(video_uri='https://www.youtube.com/watch?v=axUJRaHfgTc', ...),
#     ],
# )
```

### CODE REVIEW

File: cdp_scrapers/scraper_utils.py
Comment (dphoria) [inline cdp_scrapers/scraper_utils.py:177]: Something happened to I think `dataclasses` such that the current typing syntax used for the ingestion models, e.g. `var: str | None = None` [triggers `TypeError`](https://github.com/CouncilDataProject/cdp-scrapers/actions/runs/4335969753/jobs/7570940026#step:5:50).

This is a temporary patch that should be removed when we can just call `from_dict` again. Other instances that call `parse_static_file` will break if they update `cdp_scrapers`, but that is intentional, for the additional time zone parameter into that file.

Comment (dphoria) [reply cdp_scrapers/scraper_utils.py:177]: e.g.

```Python

from dataclasses import dataclass
from dataclasses_json import DataClassJsonMixin
from typing import Optional

@dataclass
class A(DataClassJsonMixin):
    attr_a: str = None

@dataclass
class A(DataClassJsonMixin):
    attr_a: str | None = None

>>>

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 10
      5 @dataclass
      6 class A(DataClassJsonMixin):
      7     attr_a: str = None
      9 @dataclass
---> 10 class A(DataClassJsonMixin):
     11     attr_a: str | None = None

Cell In[2], line 11, in A()
      9 @dataclass
     10 class A(DataClassJsonMixin):
---> 11     attr_a: str | None = None

TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'
```

File: cdp_scrapers/instances/portland.py
Comment (evamaxfield) [inline cdp_scrapers/instances/portland.py:41]: I know this PR isn't that place to discuss this but I am going to note here that it seems like some of the static data JSON files are out of date. I think Portland specifically but once the backfill is done I will try to compile a list to update -- I will help update them too hahah

File: cdp_scrapers/scraper_utils.py
Comment (evamaxfield) [reply cdp_scrapers/scraper_utils.py:177]: I KNOW WHAT THIS IS AND IT IS ANNOYING.

dataclass_json simply doesn't allow the python 3.11 `|` -> `Union` automatic conversion.

See this issue: https://github.com/lidatong/dataclasses-json/issues/349

I think we should maybe move to a different library than dataclasses json because of this bug (and my discovery that it is slow to decode and encode) -- that is future work. To fix this current issue you can simply change the `str | None` to `Union[str, None]`

**Review by evamaxfield — APPROVED** [review]
This is awesome. I am going to try and play with it tomorrow and see how it works but the body to search terms dict makes total sense to me.

Nice work.

If you could fix those small issues I mentioned and let me try it, we can merge later this week / weekend.

**Comment by dphoria** [comment]
Switched to draft to ensure I make all the updates I want before merging.

**Comment by dphoria** [comment]
Trivial changes since 2 weeks ago; syncing upstream and cleanup. Will wait a couple of days just in case we think of something then merge some time this weekend.

**Comment by evamaxfield** [comment]
> Trivial changes since 2 weeks ago; syncing upstream and cleanup. Will wait a couple of days just in case we think of something then merge some time this weekend.

Sounds good! You have my approval to merge whenever! THANK YOU SO MUCH ON THIS.

Also -- just want to truly express my gratitude for your work in general. Someone at the conference today after I presented was like "I wish I had transcripts for the meetings in this city we are following for a story -- like the last six months of meetings"

I asked what city it was, and it's a PrimeGov city. Deployed in ~1 hour: https://github.com/CouncilDataProject/city-of-menifee

Going to work on fixing the pipeline for 30 minutes and see if I can get them that data literally in a 1 day turn around. Your work helped do this. Thank you @dphoria

**Comment by dphoria** [comment]
> > Trivial changes since 2 weeks ago; syncing upstream and cleanup. Will wait a couple of days just in case we think of something then merge some time this weekend.
> 
> Sounds good! You have my approval to merge whenever! THANK YOU SO MUCH ON THIS.
> 
> Also -- just want to truly express my gratitude for your work in general. Someone at the conference today after I presented was like "I wish I had transcripts for the meetings in this city we are following for a story -- like the last six months of meetings"
> 
> I asked what city it was, and it's a PrimeGov city. Deployed in ~1 hour: https://github.com/CouncilDataProject/city-of-menifee
> 
> Going to work on fixing the pipeline for 30 minutes and see if I can get them that data literally in a 1 day turn around. Your work helped do this. Thank you @dphoria

For a split second I thought "Wait do we get transcripts from PrimeGove yet?" then I remembered "WHO CARES BECAUSE WE GOT WHISPER" :rofl: . Thanks for your kind words! Let me know tomorrow/weekend if we need to do some quick patches/fixes to the PrimeGov scraper get whatever data they need. I had forgotten about that for a while LOL.

**Pull request merged**
